src/Common/src/Interop/Windows/mincore/Interop.FormatMessage.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Text;
+using System.Runtime.InteropServices;
+
+internal partial class Interop
+{
+    internal partial class mincore
+    {
+        private const int FORMAT_MESSAGE_IGNORE_INSERTS = 0x00000200;
+        private const int FORMAT_MESSAGE_FROM_SYSTEM = 0x00001000;
+        private const int FORMAT_MESSAGE_ARGUMENT_ARRAY = 0x00002000;
+
+        private const int ERROR_INSUFFICIENT_BUFFER = 0x7A;
+
+        [DllImport(Libraries.Localization, CharSet = CharSet.Unicode,  EntryPoint="FormatMessageW", SetLastError = true, BestFitMapping = true)]
+        private static extern int FormatMessage(int dwFlags, IntPtr lpSource_mustBeNull, uint dwMessageId,
+            int dwLanguageId, StringBuilder lpBuffer, int nSize, IntPtr[] arguments);
+
+        /// <summary>
+        ///     Returns a string message for the specified Win32 error code.
+        /// </summary>
+        internal static string GetMessage(int errorCode)
+        {
+            string errorMsg;
+
+            var sb = new StringBuilder(256);
+            do
+            {
+                if (TryGetErrorMessage(errorCode, sb, out errorMsg))
+                    return errorMsg;
+                else
+                {
+                    // increase the capacity of the StringBuilder by 4.
+                    sb.Capacity *= 4;
+                }
+            }
+            while (sb.Capacity < MaxAllowedBufferSize);
+
+            // If you come here then a size as large as 65K is also not sufficient and so we give the generic errorMsg.
+            return string.Format("Unknown error (0x{0:x})", errorCode);
+        }
+
+        private static bool TryGetErrorMessage(int errorCode, StringBuilder sb, out string errorMsg)
+        {
+            errorMsg = "";
+
+            int result = FormatMessage(FORMAT_MESSAGE_IGNORE_INSERTS |
+                                       FORMAT_MESSAGE_FROM_SYSTEM |
+                                       FORMAT_MESSAGE_ARGUMENT_ARRAY,
+                                       IntPtr.Zero, (uint)errorCode, 0, sb, sb.Capacity + 1,
```

Comment:
You are right, it's a potential bug mentioned in item 3 #287
I just copied this code "as-is" from existing Win32Exception.GetMessage.
So should I remove this "+1" ?